### ISSUE TITLE
Definition about Job and Task

### ISSUE THREAD

**Issue by kensongzhu** [reinterpretcat/vrp#90]
Hi, 

Quite new to the topic **vrp** and still not so sure about the concepts Job and Task mentioned in the docs.
 
Say I have 2 deliveries to 2 customers with same location, same time window and same demand. 

Are there any conceptual differences with following problem descriptions?

1. Define 2 identical jobs
 
```json
[
  {
    "id": "job1",
    "deliveries": [
      {
        "places": [
          {
            "location": {
              "lat": 52.52599,
              "lng": 13.45413
            },
            "duration": 300.0,
            "times": [["2019-07-04T09:00:00Z", "2019-07-04T18:00:00Z"]]
          }
        ],
        "demand": [1]
      }
    ]
  },
 {
    "id": "job2",
    // ...rest are the same
 }
]
```

2. Define 1 job with 2 identical deliveries

```json
[
  {
    "id": "job1",
    "deliveries": [
      {
        "places": [
          {
            "location": {
              "lat": 52.52599,
              "lng": 13.45413
            },
            "duration": 300.0,
            "times": [["2019-07-04T09:00:00Z", "2019-07-04T18:00:00Z"]]
          }
        ],
        "demand": [1]
      },
      {
        // ...same delivery as above
      }
    ]
  }
]
```

3. Define 1 Job and 1delivery but doubling the demand

```json
[
  {
    "id": "job1",
    "deliveries": [
      {
        "places": [
          {
            "location": {
              "lat": 52.52599,
              "lng": 13.45413
            },
            "duration": 300.0,
            "times": [["2019-07-04T09:00:00Z", "2019-07-04T18:00:00Z"]]
          }
        ],
        "demand": [2]
      }
    ]
  }
]
```

**Comment by reinterpretcat** [comment]
Here is the differences:

1. This is two individual jobs: they are independent, so one can be assigned and another not (or both are assigned). If your vehicle has capacity [1], then only one will be assigned based on objective (e.g. minimize cost of serving it) as they have demand of [1].
2. This is one job (called internally multi-job) and it has two activities (both are deliveries). Such jobs have one extra constraint: all activities of such jobs should be assigned in the same tour. Otherwise job is considered unassigned (none of activities are assigned). If your vehicle has capacity [1], then this job will be unassigned as it requires vehicle's capacity to be at least [2]
3. This is normal (single) job with demand [2] which will have one activity (delivery) in the tour. It also requires vehicle capacity to be at least [2].

In short, amount of JobTask specifies amount of activities for a single job. All of them should be assigned or none of them.
There is also concept of JobPlace (that's why _places_ is an array) which is alternative place to serve JobTask (think about use case of delivering parcel in the morning to customer's work or, alternatively, in the evening to customer's home).

I hope this helps

**Comment by kensongzhu** [comment]
Thanks for your quick reply, which helps a lot for me to understand it. 

A further question regarding to what I wanna achieve, namely a  Multi-Depot scenario

e.g. A freight operator has homo fleet in 2 depots, which have same vehicle type, start time and end time. 

1. If define 2 **Vehicles**, but of course, it violates the **typeId** uniqueness

```json
[
   {
        "typeId": "car",
        "vehicleIds": ["car_1"],
   },
   {
        "typeId": "car",
        "vehicleIds": ["car_2"],
   }
]
```
2. If define in **shifts**, then the start time and end time must be different
```json
{
  "shifts": [
    {
      "start": {
        "earliest": "2019-07-04T08:00:00Z",
        "location": {
          "lat": 52.4664,
          "lng": 13.4023
        }
      },
      "end": {
        "latest": "2019-07-04T20:00:00Z",
        "location": {
          "lat": 52.4664,
          "lng": 13.4023
        }
      }
    },
    // second depot with different location?
    {
      "start": {
        "earliest": "2019-07-04T08:00:00Z",
        "location": {
          "lat": 52.5764,
          "lng": 13.4023
        }
      },
      "end": {
        "latest": "2019-07-04T20:00:00Z",
        "location": {
          "lat": 52.5764,
          "lng": 13.4023
        }
      }
    }
  ]
}
```

How could I describe my problem here in this case?

**Comment by kensongzhu** [comment]
Okay, it seems logical. Thanks for your suggestion
